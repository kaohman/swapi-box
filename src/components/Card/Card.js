import React from 'react';

const Card = ({ data }) => {
  const cards = data.map(card => {
    switch (card.type) {
      case 'people':
        const { homeworld, species, population } = card;
        return (
          <div>
            <h3>{card.name}</h3>
            <p>Home World: {homeworld}</p>
            <p>Species: {species}</p>
            <p>Population of Home World: { parseInt(population) ? parseInt(population).toLocaleString() : population }
            </p>
          </div>
        )
      case 'planet':
        const { terrain, climate, residents } = card;
        let residentsList = '';
        if (typeof residents === 'string') {
          residentsList = residents
        } else {
          residentsList = residents.reduce((acc, resident, i) => {
            acc = (i === residents.length - 1) ? acc + resident : acc + resident + ', ';
            return acc
          }, '');
        }
        return (
          <div>
            <h3>{card.name}</h3>
            <p>Terrain: {terrain}</p>
            <p>Climate: {climate}</p>
            <p>Residents: {residentsList}</p>
          </div>
        )
      case 'vehicle':
        const { model, vehicle_class, passengers } = card;
        return (
          <div>
            <h3>{card.name}</h3>
            <p>Model: {model}</p>
            <p>Class: {vehicle_class}</p>
            <p>Passenger Capacity: {parseInt(passengers).toLocaleString()}</p>
          </div>
        )
      default:
        return('Error: card case not valid');
    }
  });

  return (
    cards.map((card, i) => {
      return (
        <div className='card' key={data[i].name}>
          <img className='favorite-icon' alt='favorite icon'/>
          {card}
        </div>
      )
    })
  )
}

export default Card;