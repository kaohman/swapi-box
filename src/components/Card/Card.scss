h3,
p {
  font-size: 18px;
  font-family: $content-font;
}

h3 {
  font-size: 35px;
  margin: 10px 0 0 0;
  color: $header-font-color;
}


.card {
  animation: bounceAppear 450ms linear 0s 1 normal; 
  animation-fill-mode: both;
  color: white;
  background-color: #0b1013;
  box-shadow: 0 0 10px darkgrey;
  width: 250px;
  height: 300px;
  border-radius: 10px;
  padding: 15px;
  margin: auto;
}

.favorite-icon {
  background: url('./images/star-yellow.png');
  background-size: cover;
  display: block;
  margin: 0 0 0 auto;
  width: 30px;
  height: 30px;

  &:hover {
    background: url('./images/star-blue.png');
    background-size: cover;
    animation: wiggle .8s linear 0s infinite normal; 
    animation-fill-mode: forwards;
  }
}

.active-favorite {
  background: url('./images/star-blue.png');
  background-size: cover;
}

.card-icon {
  border: 1px solid hotpink;
  width: 100px;
  height: 100px;
}

.card-text-div {
  height: 240px;
  overflow: scroll;
}

@keyframes wiggle {
  0% {
    transform: rotate(0deg);
  }
  30% {
    transform: rotate(7deg);
  }
  70% {
    transform: rotate(-7deg);
  }
}

@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}