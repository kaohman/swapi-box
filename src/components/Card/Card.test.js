import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card

      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});