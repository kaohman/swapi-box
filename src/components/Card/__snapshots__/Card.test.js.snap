// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in 1`] = `
<div
  className="card"
>
  <img
    className="favorite-icon"
  />
  <h3>
    NAME HERE
  </h3>
  <img
    className="card-icon"
  />
  <p>
    Other text here
  </p>
</div>
`;
